examples/static_map/host_bulk_example.cu

```diff
@@ -57,16 +58,21 @@ int main(void)
   auto map = cuco::static_map{
     capacity, cuco::empty_key{empty_key_sentinel}, cuco::empty_value{empty_value_sentinel}};
 
-  // Create a sequence of keys and values {{0,0}, {1,1}, ... {i,i}}
+  // Create a sequence of keys and values
   thrust::device_vector<Key> insert_keys(num_keys);
   thrust::sequence(insert_keys.begin(), insert_keys.end(), 0);
   thrust::device_vector<Value> insert_values(num_keys);
   thrust::sequence(insert_values.begin(), insert_values.end(), 0);
-  auto zipped =
-    thrust::make_zip_iterator(thrust::make_tuple(insert_keys.begin(), insert_values.begin()));
+  // Combine keys and values into pairs {{0,0}, {1,1}, ... {i,i}}
+  auto pairs = thrust::make_transform_iterator(
+    thrust::counting_iterator<std::size_t>{0},
+    cuda::proclaim_return_type<cuco::pair<Key, Value>>(
+      [keys = insert_keys.begin(), values = insert_values.begin()] __device__(auto i) {
+        return cuco::pair<Key, Value>{keys[i], values[i]};
+      }));
```

Comment:
We could simplify the logic by using a transform iterator along with a counting iterator, eliminating the need to materialize both keys and values. However, I prefer to retain the materialization since this code also demonstrates how users can combine keys and values on the fly without needing a dedicated pair array.

Reply:
Yep, I totally agree with that.